app/controllers/products_controller.rb

```diff
@@ -0,0 +1,35 @@
+class ProductsController < ApplicationController
+
+  def new
+    @product = Product.new
+  end
+
+  def create
+    @product = Product.create(product_params)
+    redirect_to vendor_path(@product.vendor_id)
+  end
+
+  def destroy
+    p = Product.find(params[:id])
+    v_id = p.vendor_id
+    p.destroy
+    redirect_to vendor_path(v_id)
+  end
+
+  def edit
+    @product = Product.find(params[:id])
```

Comment:
Similarly to my comment in the `MarketsController` file, we could DRY out this controller just a bit by moving this line and the duplicates of it into a `before_action`.